ksqldb-functional-tests/src/test/resources/sql-tests/query-upgrades/rowpartition-rowoffset.sql

```diff
@@ -0,0 +1,59 @@
+----------------------------------------------------------------------------------------------------
+--@test: upgrade query across pseudocolumn versions
+----------------------------------------------------------------------------------------------------
+SET 'ksql.create.or.replace.enabled' = 'true';
+SET 'ksql.rowpartition.rowoffset.enabled' = 'false';
+
+CREATE STREAM a (id INT KEY, col1 INT, col2 INT) WITH (kafka_topic='a', value_format='JSON');
+
+CREATE STREAM b AS SELECT id, col1 FROM a EMIT CHANGES;
+
+INSERT INTO a (id, col1, col2) VALUES (1, 0, 1);
+ASSERT VALUES b (id, col1) VALUES (1, 0);
+
+SET 'ksql.rowpartition.rowoffset.enabled' = 'true';
+
+CREATE OR REPLACE STREAM b AS SELECT id, col1, col2 FROM a WHERE col1 > 0 EMIT CHANGES;
+
+INSERT INTO a (id, col1, col2) VALUES (1, 0, 1);
+INSERT INTO a (id, col1, col2) VALUES (1, 1, 1);
+
+ASSERT VALUES b (id, col1, col2) VALUES (1, 1, 1);
+
+--partition and offset don't work with SQL testing tool at the moment,
+--this just exists as a sanity check to ensure we have enabled the feature.
+CREATE STREAM c AS SELECT ID, ROWPARTITION AS rp, ROWOFFSET AS ro FROM b EMIT CHANGES;
```

Comment:
This test doesn't seem to be testing what it claims to test? As written, the test verifies that CSAS is allowed on sources that have been CREATE OR REPLACE'd. It does not verify that CREATE OR REPLACE is allowed/disallowed for ROWPARTITION and ROWOFFSET. I think the more useful tests to add in this file are to check that the user gets a good error message if they try to CREATE OR REPLACE an old query and add ROWPARTITION or ROWOFFSET.

Reply:
Good point- this was just  a sanity check that we are able to actually use ROWPARTITION//ROWOFFSET (ie the feature was turned on). That doesn't really seem great, as the test is testing itself a bit (the SET functionality), and nothing else. The test you mentioned is a lot better.